### PR TITLE
Add Api::CreditCardsController#update endpoint

### PR DESCRIPTION
**Pull request by jordan-brough**
To allow updating a user's credit cards. E.g. updating the name or
address on a credit card from the 'my account' page.

### CODE REVIEW

**Comment by athal7** [comment]
:+1:

File: core/lib/spree/core/controller_helpers/strong_parameters.rb [hunk 1]
```diff
@@ -10,6 +10,10 @@ def permitted_attributes
                  to: :permitted_attributes,
                  prefix: :permitted
 
+        def permitted_credit_card_attributes
+          permitted_source_attributes
```
Comment (Senjai) [inline core/lib/spree/core/controller_helpers/strong_parameters.rb]: We can't use this for user facing authorization.

The user can change the following:

```
[:number,
 :month,
 :year,
 :expiry,
 :verification_value,
 :first_name,
 :last_name,
 :cc_type,
 :gateway_customer_profile_id,
 :gateway_payment_profile_id,
 :last_digits,
 :name,
 :encrypted_data]
```

Comment (jordan-brough) [inline core/lib/spree/core/controller_helpers/strong_parameters.rb]: @Senjai if that's an issue then I don't think it's specific to this request is it?  Those attributes are already exposed via payments_attributes -> source_attributes on the order (when creating a payment).  Perhaps we need a separate PR for that if it's an issue?

**Comment by Senjai** [comment]
:+1:

**Comment by cbrunsdon** [comment]
:+1: . 
I have a bunch of complaints on the controller/spec's that were there before, but can't fault you for those.

**Pull request closed without merging**